app/controllers/results_submission_controller.rb

```diff
@@ -97,6 +97,75 @@ def upload_json
     end
   end
 
+  def import_from_live
+    @competition = competition_from_params
+    return redirect_to competition_submit_results_edit_path if @competition.results_submitted?
+
+    errors = []
+    results_to_import = @competition.rounds.flat_map do |round|
+      round.round_results.map do |result|
+        results_to_import >> InboxResult.new({
+                                               person_id: result["personId"],
+                                               pos: result["position"],
+                                               event_id: round.event_id,
+                                               round_type_id: round.round_type_id,
+                                               format_id: round.format_id,
+                                               best: result["best"],
+                                               average: result["average"],
+                                               value1: result["attempts"][0].result,
+                                               value2: result["attempts"] & [1].result,
+                                               value3: result["attempts"] & [2].result,
+                                               value4: result["attempts"] & [3].result,
+                                               value5: result["attempts"] & [4].result,
+                                             })
+      end
+    end
+
+    person_with_results = results_to_import.map(&:person_id).uniq
+
+    persons_to_import = @competition.registrations.wcif_ordered
+                                    .includes([:user])
+                                    .to_enum
+                                    .with_index(1)
+                                    .select { |r, registrant_id| r.wcif_status == "accepted" && person_with_results.include?(registrant_id) }
```

Comment:
Try to merge this `person_with_results.include?` check into the chain much earlier: `@competition.registrations.where(id: person_with_results).wcif_ordered...` and see whether that works

Reply:
(not sure if it has any uexpected side-effects, but at least on the surface it's more efficient

Reply:
well currently the id here is the wcif_id so that doesn't work. And I don't think we give the raw registration_id in the wcif

Reply:
Aaah the infamous WCIF order